src/gui/wallet.go

```diff
@@ -23,50 +23,55 @@ type SpendResult struct {
 	Error       string                     `json:"error,omitempty"`
 }
 
-// Spend spend coins from specific wallet
-func Spend(gateway *daemon.Gateway, walletID string, coins uint64, dest cipher.Address) *SpendResult {
+// Spend spends coins from given wallet id
+// Args:
+//  walletID    string          ID of wallet to spend from
+//  coins       uint64          amount of coins to spend
+//  dest        ciper.Address   recipient address
+// Return:
+//  balance     *wallet.BalancePair         latest balance
+//  transaction *visor.ReadableTransaction  readable transaction
+//  error       error                       error in spending the coins
+func Spend(gateway *daemon.Gateway, walletID string, coins uint64, dest cipher.Address) (balance *wallet.BalancePair, transaction *visor.ReadableTransaction, spendError error) {
 	var tx *coin.Transaction
 	var b wallet.BalancePair
 	var err error
-	for {
-		tx, err = gateway.Spend(walletID, coins, dest)
-		if err != nil {
-			break
-		}
 
-		var txStr string
-		txStr, err = visor.TransactionToJSON(*tx)
-		if err != nil {
-			break
-		}
-
-		logger.Info("Spend: \ntx= \n %s \n", txStr)
+	if tx, err = gateway.Spend(walletID, coins, dest); err != nil {
```

Comment:
write it as 

```go
tx, err := gateway.Spend(...)
if err != nil { 
    ...
    return nil, nil, err 
}
```

Remove the `var` declarations at the top

Reply:
`return nil, nil, err` ?
Should I not return the wallet balance and transaction when it fails?

Reply:
No, you should return empty values for the initial arguments, when there is an error

Reply:
No, you should return empty values for the initial arguments, when there is an error